unittests/FEXLinuxTests/tests/syscalls/syscalls_efault.cpp

```diff
@@ -0,0 +1,163 @@
+#include <catch2/catch.hpp>
+
+#include <cstdint>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+#include <poll.h>
+#include <signal.h>
+
+TEST_CASE("poll") {
+  // poll can return EFAULT if first argument is pointed to invalid pointer.
+  struct pollfd *invalid_fds = reinterpret_cast<struct pollfd *>(mmap(nullptr, sysconf(_SC_PAGESIZE), PROT_NONE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0));
+  auto ret = ::syscall(SYS_poll, invalid_fds, 1, 0);
+  REQUIRE(ret == -1);
+  CHECK(errno == EFAULT);
+}
+
+TEST_CASE("ppoll", "[!mayfail]") {
```

Comment:
Why is this one marked as `mayfail` but the others aren't? Shouldn't all of them fail since we don't emulate the kernel-internal recovery in the latest version?

Reply:
Because only this one may fail and only on 32-bit.

Reply:
Why do the tests pass otherwise, even though we don't emulate the tested property? What do the tests test?

Reply:
The 64-bit tests pass successfully because we pass the syscalls to the kernel. The 32-bit tests need the emulation, thus the EFAULT handling. If the EFAULT capturing is disabled then Catch2 will catch the SIGSEGV resulting in a fail, if the EFAULT capturing is enabling then we capture and result in a SIGABRT that Catch2 doesn't catch.

Reply:
> It will exit with SIGABRT because of the log message in SignalDelegator when it catches a fault           LogMan::Msg::AFmt("Received invalid data to syscall. Crashing now!");

You're now referring to the case I was *not* asking about. I probably misinterpreted what you meant with EFAULT capturing, and it's hard to interpret your explanation based on guesswork.

What I do understand now is why the tests pass on 64-bit regardless of whether we enable the new emulation code or not. You fully lost me on what's going on for 32-bit though. Could you try explaining again why some of the tests are marked as `mayfail` for 32-bit and why others are not, while clearly mentioning which configurations you're referring to?

Reply:
I'll make this easier for everyone. I removed the tag. The test is supposed to pass. It'll correctly pass when we return EFAULT instead of crashing.

Reply:
I don't understand what disconnect we're having here. The main feature of the revised PR is to detect if system calls are called with invalid arguments. How is it possible we have tests that intentionally trigger this faulty behavior, yet our detection logic isn't making them *all of them* fail on 32-bit? Is the detection logic really working properly?

Reply:
We're likely talking past each other because of test bifurcation. 32-bit, 64-bit, efault handling returning efault or just aborting.
Fundamentally there are three tests
- poll
- ppoll
- ppoll_time64 (This one is duplicated on 64-bit since it maps to the ppoll syscall)

Each test broken down.
- poll test
   - This one doesn't really matter, we pass the arguments untouched to the kernel where it correctly returns EFAULT for us. It's just coverage of the poll family of syscalls.

- ppoll (32-bit)
   - The pollfds and sigset_t argument are passed to kernel where it handles EFAULT for us
   - The timespec argument does `CopyFromUser`
   - If EFAULT handling will return EFAULT or abort depending on the constexpr
      - Current code it will SIGABRT which kills the entire Catch2 test process
   - Once EFAULT handling inevitably gets switched over to returning EFAULT, this test can be removed from the Known_Failures

- ppoll (64-bit)
   - Everything is passed to the kernel directly. We don't need to handle shit.

- ppoll_time64 (32-bit only because the previously mentioned syscall aliasing)
  - Same as  `ppoll (64-bit)`, Every argument is passed directly to the kernel without userspace touching it.

As you can see, our detection logic can't make them all fail, because we are using the kernel's mechanisms to handle most of the arguments.

And yes the detection logic is working properly.

Reply:
Thanks, that helps. The term "efault handling" was my main source of confusion (considering EFAULT isn't involved in FEX's new behavior at all), but with the breakdown I could understand what you're referring to.

With this explained, I would've been happy to keep the original `mayfail` tag, but it doesn't make a big difference where we declare the potential test failure.